lib/api/repository/auth_repository.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:lydiaryanfluttersurvey/api/exception/network_exceptions.dart';
+import 'package:lydiaryanfluttersurvey/api/service/auth_service.dart';
+import 'package:lydiaryanfluttersurvey/constants.dart';
+import 'package:lydiaryanfluttersurvey/env.dart';
+import 'package:lydiaryanfluttersurvey/model/request/login_request.dart';
+import 'package:lydiaryanfluttersurvey/model/response/login_response.dart';
+
+abstract class AuthRepository {
+  Future<LoginResponse> login(String username, String password);
+}
+
+class AuthRepositoryImpl implements AuthRepository {
+  final AuthService _authService;
```

Comment:
Do you plan or want to use this [DI framework](https://pub.dev/packages/get_it)?
It's the best one at the moment 🙌

Reply:
@ryan-conway How about it? It's optional 🙌

Reply:
@markgravity I added it in the latest commit, not sure why it didn't update on the conversation page 🤔  We now have `injection.dart` and the relevant classes are annotated with the appropriate annotation 😄